### PR TITLE
statistics: update some vars for internal sessions correctly

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/53972

Problem Summary:

### What changed and how does it work?

We need to always update these vars for internal sessions. Otherwise, it won't work. And we will use the wrong default value.

See more at the issue.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
修复 tidb_enable_async_merge_global_stats 和 tidb_analyze_partition_concurrency 在自动分析无法生效的问题
Fix tidb_enable_async_merge_global_stats and tidb_analyze_partition_concurrency not working in auto-analysis
```

### CODE REVIEW

**Review by fixdb — APPROVED** [review]
+1

File: pkg/statistics/handle/util/util.go [hunk 1]
```diff
@@ -105,6 +105,24 @@ func CallWithSCtx(pool SessionPool, f func(sctx sessionctx.Context) error, flags
...
+	analyzePartitionConcurrency, err := sctx.GetSessionVars().GlobalVarsAccessor.GetGlobalSysVar(variable.TiDBAnalyzePartitionConcurrency)
+	if err != nil {
+		return err
+	}
+	c, err := strconv.ParseInt(analyzePartitionConcurrency, 10, 64)
+	if err != nil {
+		return err
+	}
+	sctx.GetSessionVars().AnalyzePartitionConcurrency = int(c)
+
```
Comment (hawkingrei) [inline pkg/statistics/handle/util/util.go:125]: Add ```tidb_build_sampling_stats_concurrency``` here.

File: pkg/statistics/handle/globalstats/global_stats.go [hunk 2]
```diff
@@ -103,6 +103,7 @@ func MergePartitionStats2GlobalStats(
 	histIDs []int64,
 ) (globalStats *GlobalStats, err error) {
 	if sc.GetSessionVars().EnableAsyncMergeGlobalStats {
+		logutil.BgLogger().Info("use async merge global stats", zap.String("table", globalTableInfo.Name.L))
```
Comment (hawkingrei) [inline pkg/statistics/handle/globalstats/global_stats.go:106]: use sampling logger here.

Comment (hi-rustin) [reply pkg/statistics/handle/globalstats/global_stats.go]: Updated.

File: pkg/statistics/handle/util/util.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/statistics/handle/util/util.go:125]: img_v3_02bq_b32f3fa4-1020-4e9d-af1a-37f72a857d8g
It seems we can handle it correctly.

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**